src/main/java/org/apache/commons/lang3/time/FastDatePrinter.java

```diff
@@ -1338,51 +1339,51 @@ protected List<Rule> parsePattern() {
         final int length = pattern.length();
         final int[] indexRef = new int[1];
 
-        for (int i = 0; i < length; i++) {
+        for (int i = 0; i < length; i++) { //decision: +1
             indexRef[0] = i;
             final String token = parseToken(pattern, indexRef);
             i = indexRef[0];
 
             final int tokenLen = token.length();
-            if (tokenLen == 0) {
+            if (tokenLen == 0) { //decision: +1
                 break;
             }
 
             Rule rule;
             final char c = token.charAt(0);
 
-            switch (c) {
+            switch (c) { //decision: +24
```

Comment:
How is this 24 calculated? Ask because I'm not sure how to calculate my switch inside a loop.

Reply:
I basically calculated it as I would an if-elseif-... statement. Here the switch has 24 cases + default.